frontend/src/composables/useExecuteSQL.ts

```diff
@@ -1,64 +1,83 @@
-import { ref } from "vue";
+import { reactive } from "vue";
 import { isEmpty } from "lodash-es";
 import { useStore } from "vuex";
-import { isSelectStatement } from "../components/MonacoEditor/sqlParser";
+import { useI18n } from "vue-i18n";
+
+import {
+  parseSQL,
+  transformSQL,
+  isSelectStatement,
+  isMultipleStatements,
+} from "../components/MonacoEditor/sqlParser";
 
 const useExecuteSQL = () => {
   const store = useStore();
-  const isLoadingData = ref(false);
+  const { t } = useI18n();
+  const state = reactive({
+    isLoadingData: false,
+    isSelectStatement: false,
+  });
+
+  const notify = (type: string, title: string, description?: string) => {
+    store.dispatch("notification/pushNotification", {
+      module: "bytebase",
+      style: type,
+      title,
+      description,
+    });
+  };
 
   const execute = async () => {
     const queryStatement = store.state.sqlEditor.queryStatement;
     const selectedStatement = store.state.sqlEditor.selectedStatement;
     const sqlStatement = selectedStatement || queryStatement;
 
+    const { data } = parseSQL(sqlStatement);
+
     if (isEmpty(sqlStatement)) {
-      store.dispatch("notification/pushNotification", {
-        module: "bytebase",
-        style: "CRITICAL",
-        title: "Please input your SQL codes in the editor",
-      });
+      notify("CRITICAL", t("sql-editor.notify-empty-statement"));
       return;
     }
 
-    if (!isSelectStatement(sqlStatement)) {
-      store.dispatch("notification/pushNotification", {
-        module: "bytebase",
-        style: "CRITICAL",
-        title: "Only SELECT statements are allowed",
+    if (data !== null && !isSelectStatement(data)) {
+      store.dispatch("sqlEditor/setSqlEditorState", {
+        isShowExecutingHint: true,
       });
       return;
     }
+
+    if (isMultipleStatements(data)) {
```

Comment:
Should we allow user to execute sql when isLoadingData=true?

Reply:
add the logic when loading data